### PR TITLE
[SecurityBundle] Use Enum Nodes Instead Of Scalar

### PR DESCRIPTION
**Pull request by vadim2404**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | ~ |

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php [hunk 1]
```diff
@@ -59,14 +59,14 @@ public function getConfigTreeBuilder()
         $rootNode
             ->children()
                 ->scalarNode('access_denied_url')->defaultNull()->example('/foo/error403')->end()
-                ->scalarNode('session_fixation_strategy')->cannotBeEmpty()->info('strategy can be: none, migrate, invalidate')->defaultValue('migrate')->end()
+                ->enumNode('session_fixation_strategy')->cannotBeEmpty()->values(array('none', 'migrate', 'invalidate'))->defaultValue('migrate')->end()
```
Comment (Koc) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php]: You can use constants of `SessionAuthenticationStrategy`

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php [hunk 2]
```diff
@@ -59,14 +59,14 @@ public function getConfigTreeBuilder()
...
                 ->booleanNode('hide_user_not_found')->defaultTrue()->end()
                 ->booleanNode('always_authenticate_before_granting')->defaultFalse()->end()
                 ->booleanNode('erase_credentials')->defaultTrue()->end()
                 ->arrayNode('access_decision_manager')
                     ->addDefaultsIfNotSet()
                     ->children()
-                        ->scalarNode('strategy')->defaultValue('affirmative')->end()
+                        ->enumNode('strategy')->values(array('affirmative', 'consensus', 'unanimous'))->defaultValue('affirmative')->end()
```
Comment (ogizanagi) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php]: Same here about constants: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Security/Core/Authorization/AccessDecisionManager.php#L25

Comment (vadim2404) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php]: These lines missed at 2.3 branch.

**Comment by fabpot** [comment]
:+1:

**Comment by Tobion** [comment]
:+1:

**Comment by Tobion** [comment]
Thank you @vadim2404.

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php (hunk 1 above)
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php]: fixed in a95559858d9729 and also removed `cannotBeEmty` which is useless now

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php (hunk 2 above)
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php]: fixed in 2.6

**Pull request merged**